backend/app/routes.py

```diff
@@ -323,6 +323,72 @@ async def get_game_status():
         return jsonify({
             "answer": answer
         }), 200
+        
+    @app.route('/more_explanation', methods=['POST'])
+    async def more_explanation():
+        data = await request.get_json()
+        fen = data.get("fen")
+        question = data.get("question")
+        first_answer = data.get("first_answer")
+
+        if not fen or not question or not first_answer:
+            return jsonify({
+                "type": "invalid_request",
+                "message": "All 'fen', 'question' and 'first_answer' fields are required."
+            }), 400
+
+        if not stockfish.is_fen_valid(fen):
+            return jsonify({
+                "type": "invalid_fen_notation",
+                "message": "Invalid FEN string provided."
+            }), 422
+
+        if not llm_utils.is_string_valid(question):
+            return jsonify({
+                "type": "invalid_question",
+                "message": "Invalid question string provided."
+            }), 422
+            
+        if not llm_utils.is_string_valid(first_answer):
+            return jsonify({
+                "type": "invalid_first_answer",
+                "message": "Invalid first_answer string provided."
+            }), 422
+
+        try:
+            evaluation = stockfish.get_board_evaluation(fen)
+
+            # reset stockfish parameters
+            stockfish.reset_engine_parameters()
+
+            if evaluation is None:
+                raise StockfishException("no evaluation for the current fen")
+
+            try:
+                board_str = stockfish_utils.get_string_board(fen)
+                ask_input = {
+                    "board": board_str,
+                    "question": f"Answer with more explanation to the question: '{question}' which you answered as: '{first_answer}'?",
+                    "evaluation": evaluation
+                }
+
+                answer = coach.ask_chess_question(ask_input)
```

Comment:
in my opinion you have to modify the  input, including only the question and the answer, the previous request contains all the information.
Instead of create a new function in the main coach, you can modify the  `coach.ask_chess_question()` and check if the `board` and `evaluation` are empty and in this case you can put only as Input of the request, in self.ask() in main_coach.py, the `question` field as you created at line 371